sw/device/tests/rom_ext/rom_ext_parser_unittest.cc

```diff
@@ -0,0 +1,185 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/rom_exts/rom_ext_manifest_parser.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/testing/mock_mmio.h"
+#include "sw/device/rom_exts/manifest.h"
+
+namespace rom_ext_parser_unittest {
+namespace {
+using mock_mmio::MmioTest;
+using mock_mmio::MockDevice;
+using testing::ContainerEq;
+using testing::Each;
+using testing::Eq;
+using testing::Test;
+
+class ParserTest : public Test, public MmioTest {
+ protected:
+  mmio_region_t base_addr_ = {
+      .mock = dev().region(),
+  };
+};
+
+class IdentifierGetTest : public ParserTest {};
+
+TEST_F(IdentifierGetTest, Success) {
+  EXPECT_READ32(ROM_EXT_MANIFEST_IDENTIFIER_OFFSET, 0xa5a5a5a5);
+  EXPECT_EQ(rom_ext_identifier_get(base_addr_), 0xa5a5a5a5);
+}
+
+class SignatureGetTest : public ParserTest {
+ protected:
+  SignatureGetTest() {
+    for (uint32_t i = 0; i < ROM_EXT_IMAGE_SIGNATURE_SIZE_WORDS; ++i) {
+      src_.data[i] = std::uniform_int_distribution<uint32_t>()(gen_);
+    }
+
+    base_addr_.base = &src_.data[0];
+  }
+
+  rom_ext_signature_t src_;
+
+ private:
+  std::random_device rd;
+  std::mt19937 gen_ = std::mt19937(rd());
```

Comment:
You can just write `std::random_device()()`.

Reply:
Done.